### ISSUE TITLE
lxc-copy leaves new container's config file a mess

### ISSUE THREAD

**Issue by automated-response** [lxc/lxc#1711]
While this "issue" does not impact container functionality, it is a big pain for container management. Unlike `lxc-create` which generates clean config files that are more or less true to the template config, `lxc-copy` trashes the config file of the new container in the following ways:
* Removes line breaks
* Pulls all comment lines to the start of the config file regardless of original placement
* Makes no effort to duplicate the order or layout of the source containers config file

I am happy to work on a patch for this is others see it as an issue. I dug around in lxc-copy.c and I believe this is an artifact of how the new config is generated from values with function calls to set_config. I think most of these issues can be solved by a more thoughtful implementation with these same functions. Perhaps the `set_config*` function would need modification to accept an optional comment string (as a way to keep comments with relevant config values) worst case.

Also some standard processing/comparison between original and new config files may be sufficient to implement this fix without code changes, but just as a final processing step. I already do this wih the clone hook since it annoyed me so much.

Any thoughts on the best way to procede would be helpful.

**Comment by brauner** [comment]
Patches are always welcome, however I'm confused about what Python has to do with `lxc-copy`.

**Comment by brauner** [comment]
I can see the newline issue at least but that should be trivial to fix:

```
diff --git a/.local/share/lxc/a2/config b/.local/share/lxc/a1/config
index a6883ff..7edaed8 100644
--- a/.local/share/lxc/a2/config
+++ b/.local/share/lxc/a1/config
@@ -2,17 +2,21 @@
 # Parameters passed to the template: -d alpine
 # Template script checksum (SHA-1): a6ab286f89649338025fc8b5473e039527288017
 # For additional config options, please look at lxc.container.conf(5)
+
 # Uncomment the following line to support nesting containers:
 #lxc.include = /usr/share/lxc/config/nesting.conf
 # (Be aware this has security implications)
+
 # Container specific configuration
 # Network configuration
 # Mount entries
 # Clean-boot mount entires
+
 # Distribution configuration
 lxc.include = /usr/share/lxc/config/alpine.common.conf
 lxc.include = /usr/share/lxc/config/alpine.userns.conf
 lxc.arch = x86_64
+
 # Container specific configuration
 lxc.id_map = u 0 165536 65536
 lxc.id_map = g 0 165536 65536
@@ -31,6 +35,7 @@ lxc.network.name=eth0
 # lxc.mount.entry=/home/chb/mnt/E home/ubuntu/E none bind,optional,create=dir
 lxc.mount.entry=/dev/hugepages dev/hugepages none bind,optional,create=dir 0 0
 lxc.mount.entry = /sys/kernel/config sys/kernel/config none bind,optional,create=dir 0 0
+lxc.rootfs.path = dir:/home/chb/.local/share/lxc/a1/rootfs
+lxc.uts.name = a1
+
 # Network configuration
-lxc.rootfs.path = dir:/home/chb/.local/share/lxc/a2/rootfs
-lxc.uts.name = a2
```

**Comment by automated-response** [comment]
Yeah not python, I meant "with functions" not "with python".  Sorry, on a phone atm. 

Really the main issue I have is having any comments in the config file moved to the beginning of the file, which devoids them of context (at least in the way I document my config files, which may not be a common practice in general, isk).

**Comment by automated-response** [comment]
This bash script called from the clone hook fixes spacing:

```
#!/bin/bash

# Short name if necessary
if [[ ${#LXC_NAME} -gt 8 ]]; then
   export LXC_SHORT_NAME="${LXC_NAME:0:7}"
else
    export LXC_SHORTNAME="$LXC_NAME"
fi

# Replace veth.pair net name
sed "s/-${LXC_SRC_NAME:0:8}/-${LXC_NAME:0:8}/g" -i "$LXC_CONFIG_FILE"

# Add spacing
sed "s/^# /\n# /g" -i "$LXC_CONFIG_FILE"
```

**Comment by brauner** [comment]
Do they really get moved? Can you show a diff where this is happening? Also what version are you on?

**Comment by automated-response** [comment]
I'll do one now, I'm running stable and daily in 2 places, so let me check that before I waste more of everyone's time. Un momento.

**Comment by automated-response** [comment]
So this is fixed on 2.0.0, with the exception of the `#Container specific options`, namely `utsname` and `rootfs` options that get stuck to the end of the file and separated from the comment. Not really a big problem at all.

This is fixed for user added comments though so my grip is 98% mute. I'll send a little spacing fox commit and stfu.

Thank you!
